### PR TITLE
ECOPROJECT-615:Doc changes for updating the name of Poison Pill Operator to Self Node Remediation Operator

### PR DESCRIPTION
**Pull request by abhatt-rh**
Version(s): OpenShift 4.11+ 

Issue: https://issues.redhat.com/browse/TELCODOCS-581

Link to docs preview: http://file.emea.redhat.com/avbhatt/td-581/nodes/nodes/eco-self-node-remediation-operator.html

Additional information:
This PR addresses the doc changes required for Poison Pill Operator name change to Self Node Remediation Operator. Along with the Operator name, this PR also updates references to CRs, namespaces, kind, APIs, metadata, command line output, and any other updates that are related to the name change.

**IMPORTANT** 
Note to reviewer: To address this ticket, I have had to rename a couple of modules to accommodate a name change for an Operator. I used `git mv`  to rename the files and the commit did confirm this renaming.
However, on subsequent commits I noticed that the renaming didn't follow through. For example, it showed some old files as deleted and and the renamed files as new. The behavior wasn't consistent either since it did show some of the files as renamed. Therefore, the diff displays a false positive delta. No new modules were added as part of this PR

### CODE REVIEW

**Comment by abhatt-rh** [comment]
Hi @mshitrit and @prabinovRedhat
Request you to kindly review the PR for the name change and provide your inputs. It looks like an XL but the there hardly any content changes except for the ones mentioned in the first comment of the PR. 

Also, since the Node Health Check Operator is going GA with OCP 4.11, I think we can now update [this section](http://file.emea.redhat.com/avbhatt/td-581/nodes/nodes/eco-self-node-remediation-operator.html#about-self-node-remediation-operator_self-node-remediation-operator-remediate-nodes) to mention NHC. Let me know what you think.
@slintes feel free to provide your inputs too.

Thank you!

File: modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc [hunk 1]
```diff
@@ -3,10 +3,10 @@
 // *nodes/nodes/eco-poison-pill-operator.adoc
 
 :_content-type: PROCEDURE
-[id="configuring-machine-health-check-with-poison-pill_{context}"]
-= Configuring machine health checks to use the Poison Pill Operator 
+[id="configuring-machine-health-check-with-self-node-remediation-operator_{context}"]
```
Comment (mshitrit) [inline modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc:6]: Note that previously poison-pill_ didn't have the `operator` suffix, so unless it was initially wrong I'd expect the same for snr

File: modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc [hunk 2]
```diff
@@ -15,30 +15,30 @@ Use the following procedure to configure the machine health checks to use the Po
...
 [source,yaml]
 ----
-apiVersion: poison-pill.medik8s.io/v1alpha1
-kind: PoisonPillRemediationTemplate
+apiVersion: self-node-remediation.medik8s.io/v1alpha1
+kind: SelfNodeRemediationTemplate
 metadata:
   namespace: openshift-machine-api
-  name: poisonpillremediationtemplate-sample
+  name: <self-node-remediation-template>
```
Comment (mshitrit) [inline modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc:28]: I see `<>` was added only for the snr version and that the name format was changed.

Comment (abhatt-rh) [reply modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc:28]: Previously, we had only one template; however, now we have two remediation templates - `self-node-remediation-resource-deletion-template` and `self-node-remediation-node-deletion-template`. Therefore, I changed it to a variable

Comment (abhatt-rh) [reply modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc:28]: To clarify this further, I have added a call out [here](https://github.com/openshift/openshift-docs/pull/47363/files#diff-05eeedf4b60ce02ebcad384c025d613e8d4072e139fa414da412bef80987a63fR72). Hope that addresses the comment.

Comment (slintes) [reply modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc:28]: This procedure is about creating your own template in the openshift-machine-api namespace. So it's not related to the now two default templates in the SNR namespace (usually openshift-operators). So its fine to keep the fixed name, just with snr in it of course.
I'd suggest to add a note about the `spec.template.spec` part below though, which today can have a `remediationStrategy` field with either  `NodeDeletion` or `ResourceDeletion` value.

**Review by slintes — COMMENTED** [review]
not related to renaming things:

would it make sense to restructure docs around MHC / NHC / SNR? Today we have page about using MHC with SNR, then at a different place a page about SNR (which mentions MHC only but not NHC), and finally a page about NHC...

What about a top level "Remediation / Fencing" page? Explain what this is about, then start with MHC, introduce NHC as alternative, and finally SNR which can be used for both and even standalone...?

(woud be a bigger change, so it might be late for 4.11, but maybe it's something we can do early for 4.12?)

WDYT?
@beekhof @mshitrit @abhatt-rh

**Comment by abhatt-rh** [comment]
> not related to renaming things:
> 
> would it make sense to restructure docs around MHC / NHC / SNR? Today we have page about using MHC with SNR, then at a different place a page about SNR (which mentions MHC only but not NHC), and finally a page about NHC...
> 
> What about a top level "Remediation / Fencing" page? Explain what this is about, then start with MHC, introduce NHC as alternative, and finally SNR which can be used for both and even standalone...?
> 
> (woud be a bigger change, so it might be late for 4.11, but maybe it's something we can do early for 4.12?)
> 
> WDYT? @beekhof @mshitrit @abhatt-rh

Yes, I agree. We had documented these changes/features as they were introduced in their respective releases. I, too, have been thinking on these lines and given that we have completed the SNR related changes, NHC going GA, it seems worth to consider restructuring the content in the next/upcoming release.

File: modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc (hunk 2 above)
Comment (prabinovRedhat) [reply modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc:28]: As Marc mentioned in your own snrt you can add field of remediationStrategy:

Example:
apiVersion: self-node-remediation.medik8s.io/v1alpha1
kind: SelfNodeRemediationTemplate
metadata:
  name: self-node-remediation-resource-del-template
  namespace: openshift-machine-api
spec:
  template:
    spec:
      remediationStrategy: ResourceDeletion

File: modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc [hunk 3]
```diff
@@ -15,30 +15,30 @@ Use the following procedure to configure the machine health checks to use the Po
...
+.. To create the `SelfNodeRemediationTemplate` CR, run the following command:
 +
 [source,terminal]
 ----
-$ oc create -f <ppr-name>.yaml
+$ oc create -f <snr-name>.yaml
 ----
 
-. Create or update the `MachineHealthCheck` CR to point to the `PoisonPillRemediationTemplate` CR:
+. Create or update the `MachineHealthCheck` CR to point to the `SelfNodeRemediationTemplate` CR:
```
Comment (prabinovRedhat) [inline modules/eco-configuring-machine-health-check-with-self-node-remediation.adoc:41]: Meantime, you can't edit in 4.11 I don't see this section. Seems like a bug (I'll open)

File: modules/eco-self-node-remediation-operator-about.adoc [hunk 4]
```diff
@@ -0,0 +1,81 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes/eco-self-node-remediation-operator.adoc
+
+:_content-type: CONCEPT
+[id="about-self-node-remediation-operator_{context}"]
+= About the Self Node Remediation Operator
+
+The Self Node Remediation Operator runs on the cluster nodes and reboots nodes that are identified as unhealthy. The Operator uses the `MachineHealthCheck` controller to detect the health of a node in the cluster. When a node is identified as unhealthy, the `MachineHealthCheck` resource creates the `SelfNodeRemediation` custom resource (CR), which triggers the Self Node Remediation Operator.
```
Comment (prabinovRedhat) [inline modules/eco-self-node-remediation-operator-about.adoc:9]: ` controller to detect the health of a node `

maybe will be better to say:
` controller to detect the health of a machine`?

File: modules/eco-self-node-remediation-operator-about.adoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+The Self Node Remediation Operator minimizes downtime for stateful applications and restores compute capacity if transient failures occur. You can use this Operator regardless of the management interface, such as IPMI or an API to provision a node, and regardless of the cluster installation type, such as installer-provisioned infrastructure or user-provisioned infrastructure.
+
+[id="understanding-self-node-remediation-operator-config_{context}"]
+== Understanding the Self Node Remediation Operator configuration
+
+The Self Node Remediation Operator creates the `SelfNodeRemediationConfig` CR with the name `self-node-remediation-config` in the Self Node Remediation Operator's namespace. You can edit this CR. However, you cannot create a new CR for the Self Node Remediation Operator.
```
Comment (prabinovRedhat) [inline modules/eco-self-node-remediation-operator-about.adoc:16]: `However, you cannot create a new CR for the Self Node Remediation Operator.`
- You can create (you even have such option in the UI), but this config will be ignored.

- You will see this messafe in the logs:
```
controllers.SelfNodeRemediationConfig
ignoring selfnoderemediationconfig CRs that are not named 'self-node-remediation-config'
or not in the namespace of the operator:
'openshift-operators' {"selfnoderemediationconfig":
"openshift-operators/selfnoderemediationconfig-copy"}

```

File: modules/eco-self-node-remediation-operator-about.adoc [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+[source,yaml]
+----
+apiVersion: self-node-remediation.medik8s.io/v1alpha1
+kind: SelfNodeRemediationConfig
+metadata:
+  name: self-node-remediation-config
+  namespace: openshift-operators
+spec:
+  safeTimeToAssumeNodeRebootedSeconds: 180 <1>
+  watchdogFilePath: /test/watchdog1 <2>
```
Comment (prabinovRedhat) [inline modules/eco-self-node-remediation-operator-about.adoc:31]: This is a default value:
`watchdogFilePath: /dev/watchdog`

File: modules/eco-self-node-remediation-operator-about.adoc [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+<6> Specify a threshold value. After reaching this threshold, the node starts contacting its peers.
+<7> Specify the timeout duration for the peer to connect the API server.
+<8> Specify the timeout duration for establishing connection with the peer.
+<9> Specify the timeout duration to get a response from the peer.
+<10> Specify the frequency to update peer information, such as IP address.
+
+[id="understanding-self-node-remediation-remediation-template-config_{context}"]
+== Understanding the Self Node Remediation Template configuration
+
+The Self Node Remediation Operator also creates the `SelfNodeRemediationTemplate` CR with the name `self-node-remediation-default-template` in the Self Node Remediation Operator's namespace. This CR defines the remediation strategy for the nodes.
```
Comment (prabinovRedhat) [inline modules/eco-self-node-remediation-operator-about.adoc:58]: `The Self Node Remediation Operator also creates the `SelfNodeRemediationTemplate` CR with the name `self-node-remediation-default-template``

In 4.11 SNR will create 2 templates on for Resource Deletion strategy and another for Node Deletion strategy:
```
openshift-operators     self-node-remediation-node-deletion-template       
openshift-operators     self-node-remediation-resource-deletion-template  
```

File: modules/eco-self-node-remediation-operator-about.adoc [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+The default remediation strategy is `NodeDeletion` that removes the `node` object.
```
Comment (prabinovRedhat) [inline modules/eco-self-node-remediation-operator-about.adoc:60]: `The default remediation strategy is NodeDeletion`

From 4.11 default remediation strategy is ResourceDeletion

(And we are planing to deprecate and remove Node Deletion strategy in 4.12)
cc: @mshitrit

File: modules/eco-self-node-remediation-operator-installation-cli.adoc [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+metadata:
+    name: self-node-remediation-operator
+    namespace: self-node-remediation <1>
+spec:
+    channel: stable
+    installPlanApproval: Manual <2>
+    name: self-node-remediation-operator
+    source: redhat-operators
+    sourceNamespace: openshift-marketplace
+    package: self-node-remediation-operator
```
Comment (prabinovRedhat) [inline modules/eco-self-node-remediation-operator-installation-cli.adoc:75]: I think package name is `self-node-remediation` without `operator`

@mshitrit  Can you confirm this?

```
[kni@provisionhost-0-0 ~]$ oc get sub -A
NAMESPACE             NAME                    PACKAGE                 SOURCE          CHANNEL
openshift-operators   self-node-remediation   self-node-remediation   snr-manifests   stable
```

File: modules/eco-self-node-remediation-operator-installation-cli.adoc [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+[source,terminal]
+----
+$ oc get csv -n self-node-remediation
+----
++
+.Example output
+[source,terminal]
+----
+NAME                               DISPLAY                          VERSION   REPLACES   PHASE
+self-node-remediation.v.0.3.0      Self Node Remediation Operator   v.0.3.0              Succeeded
```
Comment (prabinovRedhat) [inline modules/eco-self-node-remediation-operator-installation-cli.adoc:100]: I think 0.4.0:
```
[kni@provisionhost-0-0 ~]$ oc get csv -n openshift-operators
NAME                           DISPLAY                          VERSION   REPLACES   PHASE
self-node-remediation.v0.4.0   Self Node Remediation Operator   0.4.0                Succeeded

```

File: modules/eco-self-node-remediation-operator-about.adoc (hunk 7 above)
Comment (mshitrit) [reply modules/eco-self-node-remediation-operator-about.adoc:58]: Good catch !

File: modules/eco-self-node-remediation-operator-about.adoc (hunk 8 above)
Comment (mshitrit) [reply modules/eco-self-node-remediation-operator-about.adoc:60]: - I agree that default  remediation strategy should be fixed (from  NodeDeletion to ResourceDeletion)
- Regarding commenting on NodeDeletion removed in 4.12 I'm not so sure - no strong opinion either way so I'm fine with following the usual convention for such use case.

File: modules/eco-self-node-remediation-operator-installation-cli.adoc (hunk 9 above)
Comment (mshitrit) [reply modules/eco-self-node-remediation-operator-installation-cli.adoc:75]: Yep :+1:

File: modules/eco-self-node-remediation-operator-about.adoc (hunk 8 above)
Comment (abhatt-rh) [reply modules/eco-self-node-remediation-operator-about.adoc:60]: Thank you @prabinovRedhat and @mshitrit 
As you might know, the updates related to the default remediation strategy and the two remediation templates were reported recently and they are now being addressed as part of TELCODOCS-790. I will share the https://github.com/openshift/openshift-docs/pull/47700 for review after it is completed.

File: modules/eco-self-node-remediation-operator-about.adoc (hunk 5 above)
Comment (abhatt-rh) [reply modules/eco-self-node-remediation-operator-about.adoc:16]: Outside the scope of this PR, needs a bug

File: modules/eco-self-node-remediation-operator-about.adoc [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+The `SelfNodeRemediationTemplate` CR resembles the following YAML file:
+
+[source,yaml]
+----
+apiVersion: self-node-remediation.medik8s.io/v1alpha1
+kind: SelfNodeRemediationTemplate
+metadata:
+creationTimestamp: "2022-03-02T08:02:40Z"
+generation: 1
+name: self-node-remediation-default-template
```
[the remark this answers is not in the document]
Comment (abhatt-rh) [reply modules/eco-self-node-remediation-operator-about.adoc:72]: Again, this comment is outside of the scope of this PR; however, these are addressed as part of TELCODOCS-790 under https://github.com/openshift/openshift-docs/pull/47700

File: modules/eco-self-node-remediation-operator-about.adoc (hunk 6 above)
Comment (prabinovRedhat) [reply modules/eco-self-node-remediation-operator-about.adoc:31]: https://bugzilla.redhat.com/show_bug.cgi?id=2107207

File: modules/eco-self-node-remediation-operator-about.adoc (hunk 4 above)
Comment (abhatt-rh) [reply modules/eco-self-node-remediation-operator-about.adoc:9]: Outside the scope of this PR, needs a bug

**Review by mjpytlak — COMMENTED** [review]
@abhatt-rh Seems like a pretty straight forward search/replace effort. With all of the renamed files, did you verify that you have not broken xrefs elsewhere in the doc set?

**Comment by abhatt-rh** [comment]
> @abhatt-rh Seems like a pretty straight forward search/replace effort. With all of the renamed files, did you verify that you have not broken xrefs elsewhere in the doc set?

AFAIK the Poison Pill/Self Node Remediation assembly is only referenced in [Node Health Check documentation](https://docs.openshift.com/container-platform/4.10/nodes/nodes/eco-node-health-check-operator.html)

**Comment by mjpytlak** [comment]
> AFAIK the Poison Pill/Self Node Remediation assembly is only referenced in [Node Health Check documentation](https://docs.openshift.com/container-platform/4.10/nodes/nodes/eco-node-health-check-operator.html)

Thank you. Peer review LGTM. Looks like you are waiting on one final review. Let me know when you need a merge.

**Comment by slintes** [comment]
lgtm

**Comment by prabinovRedhat** [comment]
lgtm

**Pull request merged**